### ISSUE TITLE
[DOC] Enhance the documentation for topological order learning methods

### ISSUE THREAD

**Issue by adam2392** [py-why/dodiscover#133]
> Another high-level comment I have is to ask if you can add some intuition to the hyper parameters of each algorithm, so that it can inform users on how to select them. Some strategies:
> 
> 1. Include some blurbs in the `Notes` section of the docstrings of the algorithms
> 2. Include an additional module, similar to `constraint_causal_discovery.rst`, which sort of gives a more user-friendly introduction to these topological ordering methods.
> 3. some combination of the above
> 
> We can also include this in a downstream PR.

_Originally posted by @francescomontagna in https://github.com/py-why/dodiscover/issues/129#issuecomment-1517362793_

**Comment by adam2392** [comment]
@francescomontagna are you still interested in submitting a PR to document the concepts behind these methods? This will help ensure more usage and proper understanding of the methods.

**Comment by francescomontagna** [comment]
Hi @adam2392 , sorry for disappearing for a while! And thanks for the PR merging and all the hard work you put in it.
So, the idea is to focus on the hyperparameters and use this as a starting point to explain the logic behind the methods, if I understand correctly. Is that it?

**Comment by adam2392** [comment]
Yes I think just a gentle introduction to the class of methods and what are some important design decisions to make. This is almost like a "user guide" for the user, such that they can properly use and cite the methods.
